diff --git a/jaxlib/mosaic/dialect/gpu/BUILD b/jaxlib/mosaic/dialect/gpu/BUILD
--- a/jaxlib/mosaic/dialect/gpu/BUILD
+++ b/jaxlib/mosaic/dialect/gpu/BUILD
@@ -28,6 +28,7 @@
 
 td_library(
     name = "mosaic_gpu_td_files",
+    visibility = ["//visibility:public"],
     srcs = ["mosaic_gpu.td"],
     includes = ["."],
     deps = [
@@ -215,6 +216,7 @@
 # Alwayslink target, used when exporting the C API from a shared library.
 cc_library(
     name = "gpu_dialect_capi_objects",
+    visibility = ["//visibility:public"],
     srcs = DIALECT_CAPI_SOURCES,
     hdrs = DIALECT_CAPI_HEADERS,
     deps = [
diff --git a/jaxlib/mosaic/gpu/BUILD b/jaxlib/mosaic/gpu/BUILD
--- a/jaxlib/mosaic/gpu/BUILD
+++ b/jaxlib/mosaic/gpu/BUILD
@@ -43,6 +43,7 @@
 
 cc_library(
     name = "mosaic_gpu_support",
+    visibility = ["//visibility:public"],
     deps = [
         ":custom_call",
         ":runtime",
@@ -239,6 +240,7 @@
 # Alwayslink target, used when exporting the C API from a shared library.
 cc_library(
     name = "mlir_capi_objects",
+    visibility = ["//visibility:public"],
     srcs = CAPI_SOURCES,
     hdrs = CAPI_HEADERS,
     deps = [
@@ -393,6 +395,7 @@
 
 cc_test(
     name = "custom_call_test",
+    visibility = ["//visibility:public"],
     srcs = ["custom_call_test.cc"],
     args = ["--heap_check="],  # TODO(b/452513034): Enable heapcheck once the bug is fixed.
     local_defines = [
